### PR TITLE
refactor: exceptionHandleTemplate

### PR DESCRIPTION
**Pull request by leizhiyuan**
fix global status is null

### Ⅱ. Does this pull request fix one issue?

fixes https://github.com/seata/seata/issues/922

image

notice！ here, when occurs exception, what the correct global status should we return? please review it. I can change it later.

### CODE REVIEW

File: server/src/main/java/io/seata/server/AbstractTCInboundHandler.java [hunk 1]
```diff
@@ -91,12 +81,42 @@ protected abstract void doGlobalCommit(GlobalCommitRequest request, GlobalCommit
...
+
+            @Override
+            public void onTransactionException(GlobalRollbackRequest request, GlobalRollbackResponse response, TransactionException tex) {
+                super.onTransactionException(request, response, tex);
+                try {
+                    GlobalSession globalSession = SessionHolder.findGlobalSession(XID.getTransactionId(request.getXid()));
+                    if (globalSession != null) {
+                        response.setGlobalStatus(globalSession.getStatus());
+                    } else {
+                        response.setGlobalStatus(GlobalStatus.UnKnown);
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/AbstractTCInboundHandler.java]: GlobalStatus.Finished

File: server/src/main/java/io/seata/server/AbstractTCInboundHandler.java [hunk 2]
```diff
@@ -91,12 +81,42 @@ protected abstract void doGlobalCommit(GlobalCommitRequest request, GlobalCommit
...
+
+            @Override
+            public void onError(GlobalRollbackRequest request, GlobalRollbackResponse response, Exception rex) {
+                super.onError(request, response, rex);
+                try {
+                    GlobalSession globalSession = SessionHolder.findGlobalSession(XID.getTransactionId(request.getXid()));
+                    if (globalSession != null) {
+                        response.setGlobalStatus(globalSession.getStatus());
+                    } else {
+                        response.setGlobalStatus(GlobalStatus.UnKnown);
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/AbstractTCInboundHandler.java]: GlobalStatus.Finished

File: server/src/main/java/io/seata/server/AbstractTCInboundHandler.java [hunk 3]
```diff
@@ -91,12 +81,42 @@ protected abstract void doGlobalCommit(GlobalCommitRequest request, GlobalCommit
...
+                    }
+                } catch (TransactionException e) {
+                    response.setGlobalStatus(GlobalStatus.UnKnown);
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/AbstractTCInboundHandler.java]: you can remove findGlobalSession's throw TransactionException, it never throw TransactionException.
image

Comment (leizhiyuan) [reply server/src/main/java/io/seata/server/AbstractTCInboundHandler.java]: ok。I will fix it

**Pull request merged**